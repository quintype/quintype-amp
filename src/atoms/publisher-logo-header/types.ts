import { LogoObj } from "./../../types/config";
import { Config } from "../../types/config";

export interface PublisherLogoHeaderTypes {
  config: Config;
  logoObj: LogoObj;
}
