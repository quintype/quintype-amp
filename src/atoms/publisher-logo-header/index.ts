import { PublisherLogoHeader } from "./publisher-logo-header";
export { PublisherLogoHeader };
