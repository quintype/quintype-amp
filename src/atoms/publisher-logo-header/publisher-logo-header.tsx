import React, { Fragment } from "react";
import { PublisherLogoHeaderTypes } from "./types";
import { withConfig } from "../../context";
import get from "lodash.get";
import { Head } from "../head";

export const PublisherLogoHeaderBase = ({ config }: PublisherLogoHeaderTypes) => {
  const publisherName = get(config, ["publisherConfig", "publisher-name"], "");
  const logo = get(config, ["ampConfig", "logo-url"], null);
  if (!logo) return null;
  return (
    <Fragment>
      <Head>
        <style>{`
          .qt-amp-publisher-logo-header img {
            object-fit: contain
          }
        `}</style>
      </Head>
      <amp-img class="qt-amp-publisher-logo-header" alt={publisherName} src={logo} layout="fill" />
    </Fragment>
  );
};

export const PublisherLogoHeader = withConfig(PublisherLogoHeaderBase);
