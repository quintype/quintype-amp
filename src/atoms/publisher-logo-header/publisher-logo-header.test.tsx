import React from "react";
import { PublisherLogoHeader } from "./publisher-logo-header";
import { shallow } from "enzyme";

describe("PublisherLogoHeader", () => {
  it("should render", () => {
    const wrapper = shallow(<PublisherLogoHeader logoSrc="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
