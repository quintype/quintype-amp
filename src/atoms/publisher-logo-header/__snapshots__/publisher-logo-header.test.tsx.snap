// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublisherLogoHeader should render 1`] = `
<a
  href="/"
>
  <amp-img
    alt="Publisher Logo"
    height="50px"
    src="https://unsplash.com/photos/ffODvgBivdw"
    width="200px"
  />
</a>
`;
